src/ngraph/runtime/backend.hpp

```diff
@@ -44,9 +44,9 @@ class ngraph::runtime::Backend
     /// \brief Create a new Backend object
     /// \param type The name of a registered backend, such as "CPU" or "GPU".
     ///   To select a subdevice use "GPU:N" where s`N` is the subdevice number.
-    /// \returns unique_ptr to a new Backend or nullptr if the named backend
+    /// \returns shared_ptr to a new Backend or nullptr if the named backend
```

Comment:
Why do you need to change this from unique_ptr to shared_ptr? If Hybrid needs a shared ptr internally - then it can get one from the unique_ptr. See this for example: https://stackoverflow.com/a/37885232

Reply:
No, you can't get a shared_ptr from a unique_ptr unless you own the unique_ptr, and the backend does not own this pointer.

Reply:
If you have hold of a `unique_ptr` you own it and can switch it to a `shared_ptr`. If you have a `unique_ptr&` then you don't own it, and a simpler approach would be to change the thing that gives you a reference to something that gives you a `unique_ptr` if it can, and, if it can't then things need to be `shared_ptr`. So, which of these scenarios are we in?

Reply:
There is a pointer that I want to shared between the backend and the frontend. In order to share the pointer it needs to be a shared_ptr. The backend used to return a raw pointer which was converted to a unique_ptr in ngraph, not in the backend.

Reply:
What's backend and what's frontend? Can you please provide a specific example?
The history of this interface is - first it was returning shared_pointer - which resulted in various issues - so it was fixed by moving to unique_ptr. Now it's moving back to shared_ptr - so I wonder whether the same issues will be back?

Reply:
Ok - based on some disussion - I now understand the reason. Looks good to me.